client/src/components/SignIn/ThirdPartyApi/Github.js

```diff
@@ -61,99 +76,128 @@ class SignInGithubBase extends Component {
-        }
-        // console.log(socialAuthUser.user.providerData["0"].uid);
-        return this.props.firebase
-          .user(socialAuthUser.user.providerData['0'].uid)
-          .set({
-            email: socialAuthUser.user.email
-          });
+        const thirdPartyUID = socialAuthUser.user.providerData['0'].uid;
+        this.setState({ uid: thirdPartyUID, isOpen: true });
       })
-      .catch((err) => {
-        if (err.code === ERROR_CODE_ACCOUNT_EXISTS) {
-          err.message = ERROR_MSG_ACCOUNT_EXISTS;
-        }
-        this.setError(err);
-      });
-    event.preventDefault();
+      .catch((err) => console.log(err));
   };
 
   render() {
     const { error } = this.state;
+    if (this.state.signedUp) {
+      return <Redirect to={ROUTES.HOME} />;
+    }
     return (
       <React.Fragment>
-        <form onSubmit={this.onSubmit}>
-          <button type="submit"> Sign In with Github </button>{' '}
-          {error && <p> {error.message} </p>}
-        </form>
-        <Modal isOpen={this.state.isOpen} contentLabel="Example Modal">
-          <div>
-            <h1>Complete Your Sign Up.</h1>
-            <Mutation mutation={firebaseSignUp}>
-              {(firebaseSignUp) => {
-                console.log({ state: this.state });
-                return (
-                  <form
-                    onSubmit={(e) => {
-                      e.preventDefault();
-                      firebaseSignUp({
-                        variables: {
-                          username: this.state.username,
-                          thirdPartyUID: this.state.uid,
-                          email: this.state.email
-                        }
-                      });
-                    }}
-                  >
-                    <input
-                      onChange={this.onChange}
-                      defaultValue={this.state.email}
-                      placeholder="email"
-                      name="email"
-                      value={this.state.email}
-                    />
-                    <input
-                      onChange={this.onChange}
-                      placeholder="username"
-                      name="username"
-                      value={this.state.username}
-                    />
-                    <button type="submit">Submit</button>
+        <Query
+          query={CHECK_IF_USER_EXISTS}
+          variables={{ thirdPartyUID: this.state.uid }}
+        >
+          {({ loading, data, error: checkError }) => {
+            if (loading)
+              return (
+                <form onSubmit={this.onSubmit}>
+                  <button type="submit" disabled>
+                    {' '}
+                    Sign In with Github{' '}
+                  </button>{' '}
+                  <div>Loading...</div>
+                  {error && <p> {error.message} </p>}
+                </form>
+              );
+            if (checkError) {
+              console.log({ error: checkError });
```

Comment:
Logs like this one is fine since we don't have a way to show errors outside the console